npm/ng-packs/packages/core/src/lib/tests/environment-utils.spec.ts

```diff
@@ -44,8 +23,58 @@ describe('EnvironmentUtils', () => {
 
   beforeEach(() => (spectator = createComponent()));
 
-  describe('#getRemoteEnv', async () => {
-    test('should call the remoteEnv URL and dispatch the SetEnvironment action ', async () => {
+  describe('#getRemoteEnv', () => {
+    const environment: Config.Environment = {
+      production: false,
+      hmr: false,
+      application: {
+        baseUrl: 'https://volosoft.com',
+        name: 'MyProjectName',
+        logoUrl: '',
+      },
+      remoteEnv: { url: '/assets/appsettings.json', mergeStrategy: 'deepmerge' },
+      oAuthConfig: {
+        issuer: 'https://api.volosoft.com',
+        clientId: 'MyProjectName_App',
+        dummyClientSecret: '1q2w3e*',
+        scope: 'MyProjectName',
+        oidc: false,
+        requireHttps: true,
+      },
+      apis: {
+        default: {
+          url: 'https://api.volosoft.com',
+        },
+      },
+    };
+
+    const customEnv = {
+      application: {
+        baseUrl: 'https://custom-volosoft.com',
+        name: 'Custom-MyProjectName',
+        logoUrl: 'https://logourl/',
+      },
+      apis: {
+        default: {
+          url: 'https://test-api.volosoft.com',
+        },
+      },
+    };
+
+    it('should call the remoteEnv URL and dispatch the SetEnvironment action for overwrite', () => {
+      setupTestAndRun({ mergeStrategy: 'deepmerge' }, deepMerge(environment, customEnv));
+    });
+
+    it('should call the remoteEnv URL and dispatch the SetEnvironment action for deepmerge', () => {
+      setupTestAndRun({ mergeStrategy: 'overwrite' }, customEnv);
+    });
+
+    it('should call the remoteEnv URL and dispatch the SetEnvironment action for customFn', () => {
+      const someEnv = { apiUrl: 'https://some-api-url' } as any;
+      setupTestAndRun({ mergeStrategy: (_, __) => someEnv }, someEnv);
+    });
```

Comment:
Can you place [test.each](https://jestjs.io/docs/en/api#1-testeachtablename-fn-timeout) here instead of these 3 tests? It will be better than that.

Reply:
Done